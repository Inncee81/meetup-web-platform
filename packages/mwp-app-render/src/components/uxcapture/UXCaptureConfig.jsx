// @flow
import React from 'react';

type callbackType = (label: string) => mixed;
type Props = {|
	onMark?: callbackType,
	onMeasure?: callbackType,
|};

export default ({ onMark, onMeasure }: Props) => {
	const config = {};

	// No need to inject `UX.config()` script if no callbacks provided
	if (!onMark && !onMeasure) {
		return null;
	}

	if (onMark) {
		config['onMark'] = onMark.toString();
	}

	if (onMeasure) {
		config['onMeasure'] = onMeasure.toString();
	}

	const uxCaptureConfigJS = `
		<script>
			if(window.UX) {
				window.UX.config(${JSON.stringify(config)});
			}
		</script>
	`;

	return (
		<div dangerouslySetInnerHTML={{ __html: uxCaptureConfigJS }} /> // eslint-disable-line react/no-danger
	);
};
