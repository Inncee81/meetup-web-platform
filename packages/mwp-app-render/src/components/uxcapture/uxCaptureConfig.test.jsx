import React from 'react';
import { shallow } from 'enzyme';
import UXCaptureConfig from './UXCaptureConfig';

const MOCK_PROPS = {
	onMark: foo => {
		if (foo) {
			return foo;
		}
	},
	onMeasure: bar => {
		if (bar) {
			return bar;
		}
	},
};

const renderComponent = (props = MOCK_PROPS) =>
	shallow(<UXCaptureConfig {...props} />);

describe('UXCaptureConfig', () => {
	it('renders correct markup when all props are defined', () => {
		expect(renderComponent()).toMatchSnapshot();
	});

	it('renders the correct markup when only onMeasure is provided', () => {
		const onMeasureProp = {
			onMeasure: MOCK_PROPS.onMeasure,
		};
		expect(renderComponent(onMeasureProp)).toMatchSnapshot();
	});

	it('renders the correct markup when only onMark is provided', () => {
		const onMarkProp = {
			onMark: MOCK_PROPS.onMark,
		};
		expect(renderComponent(onMarkProp)).toMatchSnapshot();
	});

	it('renders the correct markup when no props are provided', () => {
		expect(renderComponent({})).toMatchSnapshot();
	});

	it('renders the correct markup when unsupported props are provided', () => {
		const invalidProps = {
			...MOCK_PROPS,
			invalid: 'some prop',
		};
		expect(renderComponent(invalidProps)).toMatchSnapshot();
	});
});
