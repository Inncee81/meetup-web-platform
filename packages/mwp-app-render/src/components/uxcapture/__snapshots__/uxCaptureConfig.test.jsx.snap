// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UXCaptureConfig renders correct markup when all props are defined 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		<script>
			if(window.UX) {
				window.UX.config({
					\\"onMark\\": function onMark(foo) {
		if (foo) {
			return foo;
		}
	},
					\\"onMeasure\\": function onMeasure(bar, noop) {
		if (bar && noop) {
			return bar;
		}
	},
				});
			}
		</script>
	",
    }
  }
/>
`;

exports[`UXCaptureConfig renders the correct markup when no props are provided 1`] = `""`;

exports[`UXCaptureConfig renders the correct markup when only onMark is provided 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		<script>
			if(window.UX) {
				window.UX.config({
					\\"onMark\\": function onMark(foo) {
		if (foo) {
			return foo;
		}
	},
					
				});
			}
		</script>
	",
    }
  }
/>
`;

exports[`UXCaptureConfig renders the correct markup when only onMeasure is provided 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		<script>
			if(window.UX) {
				window.UX.config({
					
					\\"onMeasure\\": function onMeasure(bar, noop) {
		if (bar && noop) {
			return bar;
		}
	},
				});
			}
		</script>
	",
    }
  }
/>
`;

exports[`UXCaptureConfig renders the correct markup when unsupported props are provided 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "
		<script>
			if(window.UX) {
				window.UX.config({
					\\"onMark\\": function onMark(foo) {
		if (foo) {
			return foo;
		}
	},
					\\"onMeasure\\": function onMeasure(bar, noop) {
		if (bar && noop) {
			return bar;
		}
	},
				});
			}
		</script>
	",
    }
  }
/>
`;
